spec/unit/plugins/linux/network_spec.rb

```diff
@@ -1335,6 +1335,59 @@
         end
       end
 
+      describe "when there's a source field in a local route entry but no dev" do
+        let(:linux_ip_route) do
+          <<~EOM
+            default proto bird src 172.16.19.39 metric 10 \\      nexthop via 10.1.81.1 dev eth0 weight 20
+            10.116.201.0/24 dev eth0  proto kernel  src 10.116.201.76
+            192.168.0.0/24 dev eth0  proto kernel  src 192.168.0.2
+            default via 10.116.201.254 dev eth0 metric 20 src 10.116.201.74
+          EOM
+        end
+        let(:linux_ip_route_inet6) { "" }
+
+        it "maps the no-dev route and sets ipaddress" do
+          plugin.run
+          expect(plugin["ipaddress"]).to eq("172.16.19.39")
+        end
+      end
+
+      # note: default via 10.116.201.1 discarded because src not on eth0
```

Comment:
I"m not following. It has `dev eth0` in it, how do we know that 10.116.201.1 isn't the right source?  Or 172.16.19.39? How do we know that 10.116.201.74 is the "right" source?